src/pybind/mgr/crash/module.py

```diff
@@ -25,23 +103,23 @@ def handle_command(self, inbuf, command):
 
         return handler(self, command, inbuf)
 
-    @staticmethod
-    def validate_crash_metadata(inbuf):
-        # raise any exceptions to caller
-        metadata = json.loads(inbuf)
-        if 'crash_id' not in metadata:
-            raise AttributeError("missing 'crash_id' field")
-        return metadata
-
-    @staticmethod
-    def time_from_string(timestr):
+    def time_from_string(self, timestr):
         # drop the 'Z' timezone indication, it's always UTC
         timestr = timestr.rstrip('Z')
         try:
             return datetime.datetime.strptime(timestr, DATEFMT)
         except ValueError:
             return datetime.datetime.strptime(timestr, OLD_DATEFMT)
 
+    def validate_crash_metadata(self, inbuf):
+        # raise any exceptions to caller
+        metadata = json.loads(inbuf)
+        for f in ['crash_id', 'timestamp']:
+            if f not in metadata:
+                raise AttributeError("missing '%s' field" % (f))
+        time = self.time_from_string(metadata['timestamp'])
```

Comment:
I think this can be removed, doesn't look like it is used now

Reply:
we're doing the conversion to make sure the timestamp is valid